pandas/io/stata.py

```diff
@@ -1494,7 +1491,11 @@ def read(self, nrows=None, convert_dates=None,
             data = data.set_index(ix)
 
         if columns is not None:
-            data = self._do_select_columns(data, columns)
+            try:
```

Comment:
it might be worth it to create a context manager so this is a bit easier, eg.

```
with close_on_error():
    data = self._do_select_columns(data, columns)
```